src/main/scala/metagenomica/parameters.scala

```diff
@@ -38,33 +39,66 @@ trait AnyMG7Parameters {
   }
   val  blastCommand: BlastCommand
 
-  type BlastOutRec <: AnyBlastOutputRecord.For[BlastCommand]
-  val  blastOutRec: BlastOutRec
+  type BlastOutRecKeys <: AnyBlastOutputFields{
+    type Types <: AnyKList {
+      type Bound <: AnyOutputField
+      type Union <: BlastCommand#ValidOutputFields#Types#Union
+    }
+  }
+  val  blastOutRec: BlastOutputRecord[BlastOutRecKeys]
 
   val blastOptions: BlastCommand#OptionsVals
 
   val referenceDB: bundles.AnyReferenceDB
 
-  val argValsToSeq: BlastOptionsToSeq[BlastArgumentsVals] = implicitly[BlastOptionsToSeq[BlastArgumentsVals]]
-  val optValsToSeq: BlastOptionsToSeq[BlastCommand#OptionsVals] // has to be provided implicitly
+  implicit val argValsToSeq: BlastOptionsToSeq[BlastArgumentsVals] = implicitly[BlastOptionsToSeq[BlastArgumentsVals]]
+  implicit val optValsToSeq: BlastOptionsToSeq[BlastCommand#OptionsVals] // has to be provided implicitly
+
+  // The minimal set of the output fields neccessary for the MG7 generic code
+  implicit val has_qseqid:   out.qseqid.type   isOneOf BlastOutRecKeys#Types#AllTypes
+  implicit val has_sseqid:   out.sseqid.type   isOneOf BlastOutRecKeys#Types#AllTypes
+  implicit val has_bitscore: out.bitscore.type isOneOf BlastOutRecKeys#Types#AllTypes
+  implicit val has_pident:   out.pident.type   isOneOf BlastOutRecKeys#Types#AllTypes
```

Comment:
@eparejatobes if we can set `perc_identity` as an option, is this output field still necessary? (it's not used anywhere in the code, unlike the others)